geth/txqueue/fake/txservice.go

```diff
@@ -0,0 +1,40 @@
+package fake
+
+import (
+	context "context"
+	big "math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	gomock "github.com/golang/mock/gomock"
+)
+
+func NewTestServer(ctrl *gomock.Controller) (*rpc.Server, *MockFakePublicTxApi) {
+	srv := rpc.NewServer()
+	svc := NewMockFakePublicTxApi(ctrl)
+	if err := srv.RegisterName("eth", svc); err != nil {
+		panic(err)
+	}
+	return srv, svc
+}
+
+// CallArgs copied from module go-ethereum/internal/ethapi
+type CallArgs struct {
+	From     common.Address  `json:"from"`
+	To       *common.Address `json:"to"`
+	Gas      hexutil.Big     `json:"gas"`
+	GasPrice hexutil.Big     `json:"gasPrice"`
+	Value    hexutil.Big     `json:"value"`
+	Data     hexutil.Bytes   `json:"data"`
+}
+
+// FakePublicTxApi used to generate mock by mockgen util.
+// This was done because PublicTransactionPoolAPI is located in internal/ethapi module
+// and there is no easy way to generate mocks from internal modules.
+type FakePublicTxApi interface {
```

Comment:
Maybe we can call it `FakePublicTransactionPoolAPI` to make it clearer what it is?